### ISSUE TITLE
Firejail breaks Pidgin Chat Client

### ISSUE THREAD

**Issue by Xunil73** [netblue30/firejail#5764]
Since one week Pidgin doesn't start anymore with Firejail. I'm using Arch Linux 6.2.8-arch1-1 and Firejail --version 0.9.72.

The output when Pidgin is started with pidgin.profile:
~~~
Reading profile /etc/firejail/pidgin.profile
Reading profile /etc/firejail/pidgin.local
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-interpreters.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/disable-xdg.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-usr-share-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Parent pid 4564, child pid 4565
Private /etc installed in 8.20 ms
Private /usr/etc installed in 0.01 ms
Warning: /sbin directory link was not blacklisted
Warning: /usr/sbin directory link was not blacklisted
Warning: Cannot confine the application using AppArmor.
Maybe firejail-default AppArmor profile is not loaded into the kernel.
As root, run "aa-enforce firejail-default" to load it.
Child process initialized in 342.96 ms
Fontconfig error: Cannot load default config file: No such file: (null)
Pidgin 2.14.12 hat einen Speicherzugriffsfehler festgestellt und 
versucht, eine Core-Datei zu schreiben.  Dies ist ein 
Fehler im Programm und kein Fehler von Ihnen.

Wenn Sie den Absturz reproduzieren können, informieren Sie 
bitte die Entwickler mit einem Fehlerbericht auf:
https://pidgin.im/development/simpleticket/

Bitte geben Sie unbedingt an, was Sie getan haben als der 
Fehler aufgetreten ist, und posten Sie den Backtrace aus 
der Core-Datei. Falls Sie nicht wissen, wie man einen 
Backtrace erstellt, lesen Sie bitte die Informationen auf 
https://pidgin.im/development/wiki/GetABacktrace

Parent is shutting down, bye...
~~~

if it's important there's a output from `coredumpctl info /usr/bin/pidgin` attached:

[coredumpinfo_pidgin.txt](https://github.com/netblue30/firejail/files/11090281/coredumpinfo_pidgin.txt)

**Comment by Xunil73** [comment]
The same error can be found in Debian Bullseye `uname -r` `5.10.0-21-amd64` with `firejail --version` `firejail version 0.9.64.4`

theres the console output when starting Pidgin:

~~~
Reading profile /etc/firejail/pidgin.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-interpreters.inc
Reading profile /etc/firejail/disable-passwdmgr.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/disable-xdg.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-usr-share-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Warning: networking feature is disabled in Firejail configuration file
Parent pid 1484, child pid 1485
Blacklist violations are logged to syslog
Child process initialized in 373.75 ms

(Pidgin:7): dbind-WARNING **: 21:26:35.063: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-BnAkSLZNtC: Datei oder Verzeichnis nicht gefunden
libva info: VA-API version 1.10.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_10
libva error: /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so init failed
libva info: va_openDriver() returns 1
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_1_8
libva info: va_openDriver() returns 0
Pidgin 2.14.1 hat einen Speicherzugriffsfehler festgestellt und 
versucht, eine Core-Datei zu schreiben.  Dies ist ein 
Fehler im Programm und kein Fehler von Ihnen.

Wenn Sie den Absturz reproduzieren können, informieren Sie 
bitte die Entwickler mit einem Fehlerbericht auf:
http://developer.pidgin.im/simpleticket/

Bitte geben Sie unbedingt an, was Sie getan haben als der 
Fehler aufgetreten ist, und posten Sie den Backtrace aus 
der Core-Datei. Falls Sie nicht wissen, wie man einen 
Backtrace erstellt, lesen Sie bitte die Informationen auf 
http://developer.pidgin.im/wiki/GetABacktrace

Parent is shutting down, bye...
~~~

**Comment by glitsj16** [comment]
> Since one week Pidgin doesn't start anymore with Firejail. I'm using Arch Linux 6.2.8-arch1-1 and Firejail --version 0.9.72.

We'll need more context here to be able to offer any help. Does this happen without running pidgin through firejail? Does it work with --noprofile? When you (temporarily) disable all plugins, is pidgin still crashing?

> Reading profile /etc/firejail/pidgin.local

Similar remark as above. Please post this file's content here so it's clear if there's anything in it that might be causing what you're seeing.

**Comment by glitsj16** [comment]
Thanks for providing additional info. I cannot reproduce this (yet) on my Arch Linux box, but I have a hunch your issue might be caused by gstreamer. Can you add `whitelist /usr/share/gstreamer-*` and repor back if that changes anything?

**Comment by Xunil73** [comment]
The two lines did nothing. I tested it on both, Arch and Debian.

**Comment by glitsj16** [comment]
> The two lines did nothing. I tested it on both, Arch and Debian.

So much for my hunch. For now it seems you're left with commenting lines in /etc/firejail/pidgin.profile (one by one) and try to find the offending option(s) that cause the crash you're seeing...

**Comment by Xunil73** [comment]
I figured it out! Commenting out was not helpful. 
Solution: In `/etc/firejail/pidgin.profile` there is an entry `protocol unix,inet,inet6`.
Just adding `netlink` to this line made pidgin executable again. 

I noticed it after the output of `firejail --build=file.profile` has an line with just this entry:
`protocol unix,inet, inet6,netlink`.
I saved the line in an .local and it looks like it will work as before.

Thanks support! Great job!

**Comment by Xunil73** [comment]
solved

**Comment by rusty-snake** [comment]
I did not tried yet but we should at least add a comment IMHO.

**Comment by Xunil73** [comment]
`netlink` takes effect in both distros, Arch 6.2.8-arch1-1 and Firejail --version 0.9.72 and
Debian Bullseye 5.10.0-21-amd64 firejail version 0.9.64.4. In both distros only packages of the official repositories used here. On both systems the crash happend  last week, i believe there where no package upgrades of firejail or pidgin at this time. Sorry, but i have no more details or an idea whats going on there on my machine.

**Comment by glitsj16** [comment]
@Xunil73 No worries, I think it's best to add netlink in our pidgin.profile with a comment that it might work without it. Can you open a PR for that?

**Comment by kmk3** [comment]
(Offtopic)

@Xunil73 [on Mar 30](https://github.com/netblue30/firejail/issues/5764#issuecomment-1490835131):

> Debian Bullseye 5.10.0-21-amd64 firejail version 0.9.64.4.

Note that we do not maintain that version of firejail:

* <https://github.com/netblue30/firejail/blob/master/SECURITY.md>

Versions other than the latest usually have outdated profiles and may contain
bugs and security vulnerabilities that were fixed in later versions.

See the following for how to install a supported version on Debian:

* <https://github.com/netblue30/firejail#installing>

**Comment by kmk3** [comment]
> I had the same issue in openSuSE Leap 15.5 (Pidgin 2.14.8, firejail 0.9.70)
> adding `protocol netlink` to pidgin.local fixed the issue.

Note that this was fixed in firejail 0.9.72.

See also my previous [comment](https://github.com/netblue30/firejail/issues/5764#issuecomment-1494874838).
